core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -6314,15 +6311,20 @@ object ZIO extends ZIOCompanionPlatformSpecific with ZIOCompanionVersionSpecific
       case 0 => Exit.unit
       case 1 => f(as.head).unit
       case size =>
-        def worker(queue: Queue[A]): ZIO[R, E, Unit] =
-          queue.poll.flatMap {
-            case Some(a) => f(a) *> worker(queue)
-            case _       => Exit.unit
-          }
-        Queue.bounded[A](size).flatMap { queue =>
+        ZIO.suspendSucceed {
```

Comment:
This is not strictly necessary as these methods are private and always suspended, but I decided to add it to avoid potential future bugs where we accidentally call this method somewhere where it's not suspended properly